microsoft/PowerToys
# [FancyZones] Fix moving windows on other virtual desktops

## Summary of the Pull Request

Fixed virtual desktop ID usage when updating the positions of snapped windows. 
The reason was that after switching VD, `WindowCreated` was called earlier than `OnDisplayChange(DisplayChangeType::VirtualDesktop)` which updated the current virtual desktop state.


## PR Checklist

- [x] **Closes:** #21982


## Detailed Description of the Pull Request / Additional comments


## Validation Steps Performed

* Open any window on one virtual desktop.
* Open one more window of the same app on another virtual desktop and snap it to a zone.
* Return back to the previous VD and verify the first window wasn't moved.
